File: Leetcode 3701-3800/3724/3724.py
# Leetcode 3724: Minimum Operations to Transform Array
# https://leetcode.com/problems/minimum-operations-to-transform-array/
# Solved on 26th of November, 2025
class Solution:
    def minOperations(self, nums1: list[int], nums2: list[int]) -> int:
        """
        Calculates the minimum operations to transform nums1 into nums2.

        Args:
            nums1: The first list of integers.
            nums2: The second list of integers.
        Returns:
            The minimum number of operations required.
        """
        n = len(nums1)
        totalOps = 0
        minExtraOps = float('inf')
        targetLast = nums2[-1]

        for i in range(n):
            currDiff = abs(nums1[i] - nums2[i])
            totalOps += currDiff

            minVal = min(nums1[i], nums2[i])
            maxVal = max(nums1[i], nums2[i])

            dist = 0
            if targetLast < minVal:
                dist = minVal - targetLast
            elif targetLast > maxVal:
                dist = targetLast - maxVal

            if dist < minExtraOps:
                minExtraOps = dist

        return totalOps + minExtraOps + 1

File: Leetcode 3701-3800/3724/3724.java
// Leetcode 3724: Minimum Operations to Transform Array
// https://leetcode.com/problems/minimum-operations-to-transform-array/
// Solved on 26th of November, 2025
class Solution {
    /**
     * Calculates the minimum operations to transform array nums1 into nums2.
     *
     * @param nums1 The first input array.
     * @param nums2 The second input array, which is the target array.
     * @return The minimum number of operations required.
     */
    public long minOperations(int[] nums1, int[] nums2) {
        long totalOps = 0;
        long minDiff = Long.MAX_VALUE;
        int n = nums1.length;
        int lastVal = nums2[n];

        for (int i = 0; i < n; i++) {
            int curr = nums1[i];
            int target = nums2[i];
            totalOps += Math.abs(curr - target);

            int low = Math.min(curr, target);
            int high = Math.max(curr, target);

            long currentDiff = 0;
            if (lastVal < low) {
                currentDiff = low - lastVal;
            } else if (lastVal > high) {
                currentDiff = lastVal - high;
            } else {
                currentDiff = 0;
            }

            if (currentDiff < minDiff) {
                minDiff = currentDiff;
            }
        }

        return totalOps + minDiff + 1;
    }
}